File: pages/api/scrape.js
import { scrape_to_csv } from '../../scraper';

export default async function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  const { url } = req.body;

  if (!url) {
    return res.status(400).json({ error: 'URL is required' });
  }

  try {
    const csvData = await scrape_to_csv(url);
    
    res.setHeader('Content-Type', 'text/csv');
    res.setHeader('Content-Disposition', 'attachment; filename=portfolio_companies.csv');
    res.status(200).send(csvData);
  } catch (error) {
    console.error('Scraping error:', error);
    res.status(500).json({ error: 'Failed to scrape companies' });
  }
}

File: pages/api/python/scrape.py
from http.server import BaseHTTPRequestHandler
import json
import sys
import os

# Add the root directory to Python path so we can import scraper
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))))
from scraper import scrape_to_csv

def handler(request):
    """
    Vercel serverless function handler
    """
    if request.get('method', '') != 'POST':
        return {
            'statusCode': 405,
            'body': json.dumps({'error': 'Method not allowed'})
        }

    try:
        # Parse the request body
        body = json.loads(request.get('body', '{}'))
        url = body.get('url')
        
        if not url:
            return {
                'statusCode': 400,
                'body': json.dumps({'error': 'URL is required'})
            }
            
        # Get CSV data using the scraper
        csv_data = scrape_to_csv(url)
        
        # Return the CSV data
        return {
            'statusCode': 200,
            'headers': {
                'Content-Type': 'text/csv',
                'Content-Disposition': 'attachment; filename=portfolio_companies.csv'
            },
            'body': csv_data
        }
        
    except Exception as e:
        return {
            'statusCode': 500,
            'body': json.dumps({'error': str(e)})
        }